Explorer/Assets/Scripts/ECS/StreamableLoading/Common/Systems/ReportStreamableLoadingErrorSystem.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Arch.Core;
+using Diagnostics.ReportsHandling;
+using ECS.Abstract;
+using ECS.StreamableLoading.Common.Components;
+using System;
+using UnityEngine;
+
+namespace ECS.StreamableLoading.Common.Systems
+{
+    /// <summary>
+    ///     Reads promises and reports their exceptions.
+    ///     <para>It's very important to update this system right after the system that creates <see cref="StreamableLoadingResult{TAsset}" /> as promises are consumed in a closest update cycle</para>
+    /// </summary>
+    public abstract class ReportStreamableLoadingErrorSystem<TIntention, TAsset> : BaseUnityLoopSystem
+        where TIntention: struct, IAssetIntention
+    {
+        private const LogType LOG_TYPE = LogType.Exception;
+
+        private readonly IReportsHandlingSettings settings;
+
+        private static readonly QueryDescription QUERY = new QueryDescription()
+                                                        .WithAll<TIntention, StreamableLoadingResult<TAsset>>()
+                                                        .WithNone<LoadingInProgress>();
+
+        private TryReport tryReport;
+
+        protected ReportStreamableLoadingErrorSystem(World world, IReportsHandlingSettings settings) : base(world)
+        {
+            this.settings = settings;
+            tryReport = new TryReport(GetReportCategory());
+        }
+
+        protected override void Update(float t)
+        {
+            // Skip the update entirely if the category is disabled
+            if (!settings.CategoryIsEnabled(GetReportCategory(), LOG_TYPE))
+                return;
+
+            World.InlineQuery<TryReport, StreamableLoadingResult<TAsset>>(in QUERY, ref tryReport);
```

Comment:
How does this not loop forever, is there another system that removes it?

Reply:
The flow expects that the promise is consumed later this frame or earlier next frame. This is by design.
Repetitive reports should indicate that there is a hole insystems.
I will address it in another branch.